compiler/util/files.cpp

```diff
@@ -635,12 +635,16 @@ void codegen_makefile(fileinfo* mainfile, const char** tmpbinname, bool skip_com
   // Generate one variable containing all envMap information to pass to printchplenv
   for (std::map<std::string, const char*>::iterator env=envMap.begin(); env!=envMap.end(); ++env)
   {
-    if(!useDefaultEnv(env->first)) {
-      chplmakeallvars += env->first + "=" + std::string(env->second) + " ";
-    }
+    const std::string& key = env->first;
+    const char* oldPrefix = "CHPL_";
+    const char* newPrefix = "CHPL_MAKE_";
+    INT_ASSERT(key.substr(0, strlen(oldPrefix)) == oldPrefix);
+    std::string keySuffix = key.substr(strlen(oldPrefix), std::string::npos);
+    std::string chpl_make_key = newPrefix + keySuffix;
```

Comment:
Seems a little unfortunate to do the manual replacement instead of just using `printchplenv --make` (if we change the prefix or something we have 2 points of change), but since `printchplenv` takes a non-trivial amount of time to run, I can see the desire for this.

Also, is this code change just to prevent running `printchplenv` at generated code `make` time? If it is, can we split this into a different PR?

Reply:
@ronawho - it could be a separate PR, but then we'd have to do more workarounds in this PR, since we have to adjust how CHPL_MAKE_ALL_VARS is computed anyway (since it might now include variables with spaces). So it's not quite purely an optimization. Nonetheless we could do the adjustment to CHPL_MAKE_ALL_VARS here and then make this change in another PR if that's important to you. I just think it's extra work, in this situation, since I'm pretty convinced it's "better" to run printchplenv once if we can.

Reply:
Agreed that we want to only run printchplenv once -- I'm guessing this PR is already going to blow up nightly testing, so I'd prefer to minimize whatever changes we can, but if it's hard to do in this case, I'm fine with leaving it in.